clients/src/test/java/org/apache/kafka/common/requests/RequestResponseTest.java

```diff
@@ -930,6 +930,28 @@ public void testCreateTopicRequestV3FailsIfNoPartitionsOrReplicas() {
         assertTrue(exception.getMessage().contains("[foo, bar]"));
     }
 
+    @Test
+    public void testDeleteTopicsRequestNumTopics() {
+        for (short version : DELETE_TOPICS.allVersions()) {
+            DeleteTopicsRequest request = createDeleteTopicsRequest(version);
+            DeleteTopicsRequest serializedRequest = DeleteTopicsRequest.parse(request.serialize(), version);
+            // createDeleteTopicsRequest sets 2 topics
+            assertEquals(2, request.numberOfTopics());
+            assertEquals(2, serializedRequest.numberOfTopics());
```

Comment:
I suppose that this will fail if version >= 6, isn't it?

Reply:
This shouldn't fail. versions greater or equal to 6 will get the topics written to the topics field when building, and this is what numberOfTopics looks at when version is greater or equal to 6.

Reply:
Ah, right. I completely missed that part in the builder. We usually put unit tests for requests in their own classes (e.g. DeleteTopicsRequestTest) except the ones really verifying the serialisation which are here. I wonder if this one wouldn't be better in `DeleteTopicsRequestTest`. What do you think?

Slightly related, do we have a unit test verifying the normalization logic in the build?

Reply:
I put it here because of the serialization code. (Which makes the topicNames field empty in versions 6+) But I can move it there too. 
I don't think there is such a test for normalization logic, so I can add that too.

Reply:
I think the reason I didn't put it in DeleteTopicsRequestTest is also because that is an integration test. Would it make sense to rename that and create a unit test called DeleteTopicsRequestTest?

Reply:
Yeah, we can add a DeleteTopicsRequestTest in the clients package for the unit tests.

As you said, DeleteTopicsRequestTest in core package is for the integration tests of the DeleteTopics API.

Reply:
Right. That makes sense. Will do.